great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1083,9 +1083,14 @@ def _validator(self) -> V1Validator:
             raise ValueError(  # noqa: TRY003
                 "We can't validate batches that are attached to datasources without a data context"
             )
-        batch_definition = self.data_asset.add_batch_definition(
-            name="-".join([self.datasource.name, self.data_asset.name, str(uuid.uuid4())])
+
+        # note: batch definition is created but NOT added to the asset, as it should not persist
+        batch_definition = BatchDefinition(
```

Comment:
While it might feel strange to create a BatchDefinition outside of the fluent workflow, I think it demonstrates that this decoupling was a good way to implement this.